central/administration/usage/datastore/securedunits/datastore_impl.go

```diff
@@ -44,25 +44,14 @@ func (ds *dataStoreImpl) Walk(ctx context.Context, from time.Time, to time.Time,
 	}
 
 	pagination := search.NewPagination().
-		AddSortOption(search.NewSortOption(search.AdministrationUsageTimestamp)).Limit(page)
+		AddSortOption(search.NewSortOption(search.AdministrationUsageTimestamp))
 
 	query := search.NewQueryBuilder().AddTimeRangeField(
 		search.AdministrationUsageTimestamp, from, to).WithPagination(pagination).ProtoQuery()
 
-	for offset := 0; ; offset += page {
-		pagination.Offset(int32(offset))
-		units, err := ds.store.GetByQuery(ctx, query)
-		if err != nil {
-			return errors.Wrap(err, "failed to walk through usage data")
-		}
-		for _, u := range units {
-			if err := fn(u); err != nil {
-				return errors.Wrap(err, "error while processing usage data")
-			}
-		}
-		if len(units) < page {
-			break
-		}
+	err := ds.store.WalkByQuery(ctx, query, fn)
+	if err != nil {
+		return errors.Wrap(err, "error while processing usage data")
```

Comment:
```suggestion
		return errors.Wrap(err, "error while walking through usage data")
```

Reply:
Not this PR: consider implementing a graceful iteration interruption, returned by the `fn`. E.g. `ErrStopIteration`.